### ISSUE TITLE
Programmatically updating the viewer options

### ISSUE THREAD

**Issue by Gijsjan** [openseadragon/openseadragon#1817]
Hi! Thanks for a great product! It's a life saver.

I was trying to figure out how to adjust the initial options after the viewer has been created. I have found [this comment](https://github.com/openseadragon/openseadragon/issues/1312#issuecomment-330308341) and it works. I'm using the quit new Typescript typings and it doesn't support these internal (?) props. Dynamically changing the options is also not document anywhere that I can find. Is this (per the comment) the "official" way of dynamically changing the viewer options?

If yes, should/could there be a documented method `setOption` or `updateOption`? And maybe the options should/could be added to an `options` map instead of directly on to the Viewer object?

**Comment by iangilman** [comment]
Yeah, this aspect of the API is kind of a mess. Many of the options can't be modified later, while some can. I agree it would be better to have a consistent API for all of them! Of course in some cases it might not be possible (or at least very difficult) to update things after they've gotten rolling, but we can document those on a case-by-case basis (perhaps including an error in the console).

I like the idea of an explicit function, like `setOption`; this allows us to call whatever sorts of update mechanisms are required (without adding getters/setters to the API, which we currently have none of).

Making this work for every option seems like a big task, but it shouldn't be too hard to get started on… Perhaps the function can output some sort of "this is a work in progress" error if you try to set an option that's not yet configured to work.

Would you be up for taking a whack at it?

**Comment by LeDuyQuan** [comment]
Hi guys, has this enhancement function been implemented yet?

**Comment by iangilman** [comment]
@LeDuyQuan it has not. It's available if you're interested in taking it on :)
